anvil/src/main/kotlin/trikita/anvil/CustomDslSetter.kt

```diff
@@ -10,13 +10,14 @@ import android.text.TextWatcher
 import android.util.TypedValue
 import android.view.*
 import android.widget.*
-import trikita.anvil.*
 import java.util.*
 
 // weight constants
-const val FILL = ViewGroup.LayoutParams.MATCH_PARENT
-const val MATCH = ViewGroup.LayoutParams.MATCH_PARENT
-const val WRAP = ViewGroup.LayoutParams.WRAP_CONTENT
+sealed class Size {
+    object MATCH : Size()
+    object WRAP : Size()
+    class DIP(val value: Int) : Size()
```

Comment:
This kind of duality between "size" dips and "normal" dips is embarrassing. Any thoughts on that?

Reply:
can't we have Dp, Sp and Px as subclasses of Size all evaluating to Px at the end?